libs/cart/src/drivers/magento/transforms/outputs/cart.service.ts

```diff
@@ -113,14 +114,32 @@ export class DaffMagentoCartTransformer {
   }
 
   private transformShippingMethods(cart: MagentoCart): {available_shipping_methods: DaffCart['available_shipping_methods']} {
-    return {
-      available_shipping_methods: cart.shipping_addresses[0] && cart.shipping_addresses[0].available_shipping_methods
-        ? cart.shipping_addresses[0].available_shipping_methods.map(method =>
-          this.shippingRateTransformer.transform(method)
-        )
-        : []
-    }
-  }
+		if(!cart.shipping_addresses[0]) return { available_shipping_methods: [] };
+
+		const availableMethods = cart.shipping_addresses[0] && cart.shipping_addresses[0].available_shipping_methods
+		? cart.shipping_addresses[0].available_shipping_methods.map(method =>
+				this.shippingRateTransformer.transform(method)
+			)
+		: [];
+
+		return {
+			available_shipping_methods: cart.shipping_addresses[0].selected_shipping_method &&
+			!this.containsShippingMethod(cart.shipping_addresses[0].available_shipping_methods, cart.shipping_addresses[0].selected_shipping_method) 
+			? [
+					...availableMethods,
+					this.shippingRateTransformer.transform(cart.shipping_addresses[0].selected_shipping_method)
```

Comment:
I would store `cart.shipping_addresses[0].selected_shipping_method` in a local var since you reference it 3 times in this return expression.

Reply:
done